src-static/update_location_info.R

```diff
@@ -121,11 +130,14 @@ df_names <- df_ocha_names |>
   ) |>
   dplyr$transmute(
     iso3,
-    country,
-    unhcr_region = countrycode$countrycode(
-      sourcevar = iso3,
-      origin = "iso3c",
-      destination = "unhcr.region"
+    location,
+    location_note,
+    unhcr_region = suppressWarnings(
```

Comment:
What warnings are we suppressing here?

Reply:
I'll do it properly was being a bit lazy. Basically there is an annoying thing where `ifelse()` is vectorized, so can be a nice way to merge two function calls that return vectors based on some logic. In this case, I was only calling `countrycode$countrycode()` if I knew I didn't have a custom region defined, because the custom region was set to deal with missing values in `countrycode()`! But, the way `ifelse` works, both of the potential outputs are evaluated in full, and it's only the return values that the boolean is applied to. So this generate warnings that there was no match to some of the ISO3 codes passed in, which we could safely ignore since we were dealing with those elsewhere. But to be cleaner I've instead parsed the code so we only pass in subsets of the vectors we know won't generate warnings either side.

```r
# don't use dplyr to avoid warnings for missing values in countrycode
uses_custom_reg <- !is.na(df_regions$region_custom)
df_regions$unhcr_region <- NA_character_
df_regions$unhcr_region[uses_custom_reg] <- df_regions$region_custom[uses_custom_reg]
df_regions$unhcr_region[!uses_custom_reg] <- countrycode$countrycode(
  sourcevar = df_regions$iso3[!uses_custom_reg],
  origin = "iso3c",
  destination = "unhcr.region"
)
```